canopy/canopyjs/src/transactions/signing.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2019 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import crypto from 'crypto';
+import secp256k1 from 'secp256k1';
+
+function toHex(buffer: Uint8Array): string {
+  return Array.from(buffer)
+    .map(b => b.toString(16).padStart(2, '0'))
+    .join('');
+}
+
+interface IContext {
+  sign: (message: Uint8Array, privateKey: PrivateKey) => string;
+  getPublicKey: (privateKey: PrivateKey) => PublicKey;
+}
+
+interface IPrivateKey {
+  asHex: () => string;
+  asBytes: () => Buffer;
+}
+
+interface IPublicKey {
+  publicKey: Uint8Array;
+  asHex: () => string;
+  asBytes: () => Buffer;
+}
+
+class Context implements IContext {
+  sign(message: Uint8Array, privateKey: PrivateKey): string {
+    const hash = crypto
+      .createHash('sha512')
+      .update(message)
+      .digest();
+
+    const result = secp256k1.sign(hash, Buffer.from(privateKey.asBytes()));
```

Comment:
Shouldn't have any secp256k1 stuff in the abstract implementation. Looks like this is the same code as the `Secp256k1Context`?